File: src/FestivalPOS/src/api/serving.ts
import { DateTime } from "luxon";
import { dateTimeConverter } from "./converters/date-time-converter";
import { jsonProperty } from "@utiliread/json";

export class ServingLine {
    @jsonProperty()
    id!: number;

    @jsonProperty()
    servingId!: number;

    @jsonProperty()
    quantity!: number;

    @jsonProperty()
    name!: string;
}

export type ServingState = "pending" | "ongoing" | "completed";

export class Serving {
    @jsonProperty()
    id!: number;

    @jsonProperty()
    orderId!: number;

    @jsonProperty()
    pointOfSaleId!: number;

    @jsonProperty()
    state!: ServingState;

    @jsonProperty()
    highPriority!: boolean;

    @jsonProperty()
    tagNumber?: number;

    @jsonProperty({ converter: dateTimeConverter })
    created!: DateTime;

    @jsonProperty()
    staffNumber?: number;

    @jsonProperty({ converter: dateTimeConverter })
    accepted?: DateTime;

    @jsonProperty({ converter: dateTimeConverter })
    completed?: DateTime;

    @jsonProperty({ type: ServingLine })
    lines!: ServingLine[];
}

File: src/FestivalPOS/Models/Serving.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace FestivalPOS.Models
{
    public class Serving
    {
        public int Id { get; private set; }
        public int OrderId { get; set; }
        [JsonIgnore]
        public Order Order { get; private set; }
        public int PointOfSaleId { get; set; }
        [JsonIgnore]
        public PointOfSale PointOfSale { get; set; }
        public ServingState State { get; set; }
        public bool HighPriority { get; set; }
        public int? TagNumber { get; set; }
        public DateTimeOffset Created { get; set; }
        public int? StaffNumber { get; set; }
        public DateTimeOffset? Accepted { get; set; }
        public DateTimeOffset? Completed { get; set; }
        public List<ServingLine> Lines { get; set; } = new List<ServingLine>();
    }

    public enum ServingState
    {
        Pending = 1,
        Ongoing = 2,
        Completed = 3
    }
}
